app/src/main/java/com/darshan/notificity/AppDatabase.kt

```diff
@@ -25,13 +26,40 @@ abstract class AppDatabase : RoomDatabase() {
                         AppDatabase::class.java,
                         Constants.dbName
                     )
-                        .fallbackToDestructiveMigration()
+                        .addMigrations(MIGRATION_1_2)
                         .build()
                     INSTANCE = instance
                 }
                 return instance
             }
         }
+
+        /**
+        * MIGRATION 1 -> 2
+        * Change in PrimaryKey for [Notification]
+         * The primary key is now composed of [Notification.id] && [Notification.packageName]
+         * The [Notification.id] is now set to the notificationId which was originally sent to the NotificationManager
+        * */
+        private val MIGRATION_1_2: Migration = object : Migration(1, 2) {
+            override fun migrate(database: SupportSQLiteDatabase) {
+                database.execSQL("CREATE TABLE IF NOT EXISTS `notification_new` " +
```

Comment:
@zzjjaayy Can we not create a new table and update the schema for the existing Notification table and update the primary keys?

Reply:
@darshanpania I think we cannot, sqlite doesn't let you change the primary keys like that. I did try and could not get it to work.

Reply:
Checkout this https://stackoverflow.com/questions/16900552/change-the-primary-key-of-a-table-in-sqlite

Reply:
thanks @zzjjaayy and @mrwhoknows55 for the info!